dask/dataframe/tests/test_shuffle.py

```diff
@@ -304,6 +304,30 @@ def test_set_index_3(shuffle_method):
     assert ddf2.npartitions == ddf.npartitions
 
 
+def test_set_index_no_sort():
```

Comment:
I think it would be good to test the behavior of the propagated params like `drop` with `sort=False`. This might be done in explicit tests or by parametrizing existing tests, I defer this decision to you.